tools/dbtool.py

```diff
@@ -579,96 +596,135 @@ def menu():
           '|' + colorama.Fore.GREEN + 's' + colorama.Style.RESET_ALL + '. Settings                            ' + colorama.Fore.RED + '|\n'
           '|' + colorama.Fore.GREEN + 'q' + colorama.Style.RESET_ALL + '. Quit                                ' + colorama.Fore.RED + '|\n'
           + colorama.Fore.GREEN + 'o' + colorama.Fore.RED + '---------------------------------------' + colorama.Fore.GREEN + 'o')
 
 def settings():
     fetch_configs()
-    print(colorama.Fore.GREEN + 'Current MySQL bin location: ' + colorama.Style.RESET_ALL + mysql_bin)
+    print_green('Current MySQL bin location: ' + colorama.Style.RESET_ALL + mysql_bin)
     if input('Change this location? [y/N] ').lower() == 'y':
         adjust_mysql_bin()
-    print(colorama.Fore.GREEN + 'Automatic backup for command line updates: ' + colorama.Style.RESET_ALL + str(bool(auto_backup)))
+    print_green('Automatic backup for command line updates: ' + colorama.Style.RESET_ALL + str(bool(auto_backup)))
     if input('Change this? [y/N] ').lower() == 'y':
         adjust_auto_backup()
-    print(colorama.Fore.GREEN + 'Automatic client version update for command line updates: ' + colorama.Style.RESET_ALL + str(auto_update_client))
+    print_green('Automatic client version update for command line updates: ' + colorama.Style.RESET_ALL + str(auto_update_client))
     if input('Change this? [y/N] ').lower() == 'y':
         adjust_auto_update_client()
     write_configs()
 
 # TODO: Hook this up to a menu option
 def set_external_ip(ip_str):
-    command = '"' + mysql_bin + 'mysqladmin' + exe + '" -h ' + host + ' -P ' + str(port) + ' -u ' + login + ' -p' + password + \
-        'UPDATE zone_settings SET zoneip = ' + ip_str + ';'
-    os.system(command + log_errors)
+    global mysql_bin, exe
+
+    query = f'UPDATE zone_settings SET zoneip = \'{ip_str}\';'
+
+    print("Executing query:")
+    print(query)
+
+    result = subprocess.run([
+        '{}mysql{}'.format(mysql_bin, exe),
+        '-h', host,
+        '-P', str(port),
+        '-u', login,
+        '-p{}'.format(password),
+        database,
+        '-e', query],
+        capture_output=True, text=True)
+
+    for line in result.stderr.splitlines():
+        # Safe to ignore this warning
+        if 'Using a password on the command line interface can be insecure' not in line:
+            print_red(line)
+
+def tasks():
+    if input('Make server public-facing? [y/N] ').lower() == 'y':
+        import urllib.request
+        external_ip = urllib.request.urlopen('https://ident.me').read().decode('utf8')
+        print_green(f'Detected your public-facing IP as: {external_ip}')
+        if input('Is this correct? [y/N] ').lower() == 'y':
+            set_external_ip(external_ip)
+        else:
+            external_ip = input('Please enter your public-facing IP: ')
+            set_external_ip(external_ip)
```

Comment:
Should maybe validate that that `external_ip` isn't blank and skip setting if it is in case they decide to bail.